Source/Randomizer.cpp

```diff
@@ -436,12 +439,14 @@ void Randomizer::Randomize(std::vector<int>& panels, int flags) {
     return RandomizeRange(panels, flags, 0, panels.size());
 }
 
-int Randomizer::SwapWithRandomPanel(int panel1, const std::vector<int>& possible_panels, int flags) {
-    const int target = Random::RandInt(0, static_cast<int>(possible_panels.size()) - 1);
-    if (panel1 != possible_panels[target]) {
-        SwapPanels(panel1, possible_panels[target], flags);
+void Randomizer::SwapWithRandomPanel(int panel1, const std::vector<int>& possible_panels, int flags) {
+    std::vector<int> filtered = copyWithoutElements(possible_panels, _alreadySwapped);
```

Comment:
This seems excessive. Why not just re-roll if you select a vallue from the "alreadySwapped" list?
This change is going to break the seed anyways.

Reply:
Or, you could make a copy and then call `.erase()` after key panels are randomized.

Reply:
Fixed.